analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ProvideDeserializationMethodsForOptionalFields.CSharp12.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+class AttributeOnPrimaryConstructorParameter
+{
+    [Serializable]
+    public class ParameterNotUsedInProperty([OptionalField] int optionalField) // Error [CS0592]
```

Comment:
Since this is invalid, I am not convinced that we need this UT.

Reply:
After having tried that out, I have added it as UT thinking that, while the compiler doesn't currently allow to specify attributes with target `AttributeTargets.Field`, but only with `AttributeTargets.Parameter`, it may do so in the future. So I wanted to keep it there the UT as sentinel for future evolutions of this feature.

However, on a more attentive analysis, I don't see how this can possibly work:
- the parameter is either only a parameter or both a parameter and a field, depending on whether it is only used at construction time, or also later on
- so, it's quite difficult and very contextual to determine which attribute would apply to what (parameter, unspeakable field, both)

In conclusion, I have to agree with you. I have removed the test.